packages/mui-material/src/internal/SwitchBase.test.js

```diff
@@ -436,20 +436,21 @@ describe('<SwitchBase />', () => {
   });
 
   describe('checkbox form submission', () => {
-    it('dont set a void string as value', () => {
+    it('`value` falls back to the platform default if no `value` is set', () => {
       const { getByRole } = render(
         <SwitchBase icon="unchecked" checkedIcon="checked" type="checkbox" />,
       );
 
-      expect(getByRole('checkbox')).not.to.have.property('value', '');
+      // https://html.spec.whatwg.org/multipage/input.html#dom-input-value-default-on
+      expect(getByRole('checkbox')).to.have.property('value', 'on');
```

Comment:
negative assertions are good when we don't care about what a value is. But here we actually do care because `' '` should fail as well.